Sources/core/objects/Gun.cpp

```diff
@@ -0,0 +1,67 @@
+#include "Gun.h"
+
+Gun::Gun(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+
```

Comment:
Also you have introduced two new variable in `Gun` class (`max_ammunition_` and `ammunition_`), but you do not set them up in constructor. I understand that you set them in successor classes, but you should still set some default values in the base class. Because otherwise someone can forgot to make it in a some new successor class and it will cause an uninitialized variable existance (and it is really bad).